tests/testthat/test-shinytest2-init.R

```diff
@@ -56,3 +56,58 @@ testthat::test_that("e2e: init creates UI containing specified title, favicon, h
   )
   app$stop()
 })
+
+testthat::test_that("e2e: teal app initializes with Show R Code modal", {
+  app <- TealAppDriver$new(
+    data = simple_teal_data(),
+    modules = example_module(label = "Example Module")
+  )
+  app$wait_for_idle(timeout = default_idle_timeout)
+
+  # Check if button exists.
+  button_selector <- sprintf("#%s-rcode-button", app$active_module_ns())
+  testthat::expect_equal(
+    app$get_text(button_selector),
+    "Show R code"
+  )
+
+  app$click(selector = button_selector)
+  app$wait_for_idle(timeout = default_idle_timeout)
+
+  # Check header and title content.
+  testthat::expect_equal(
+    app$get_text("#shiny-modal > div > div > div.modal-header > h4"),
+    "Example Code"
+  )
+  testthat::expect_equal(
+    app$get_text(sprintf("#%s-rcode-copy_button1", app$active_module_ns())),
+    "Copy to Clipboard"
+  )
+  testthat::expect_equal(
+    app$get_text("#shiny-modal > div > div > div.modal-body > div > button:nth-child(2)"),
```

Comment:
@averissimo would you be able to help changing this `nth-child(2)` to something based on `data` property?

Reply:
You can use:
- `[<attribute_name>]` has attribute (regardless of value)
- `[<attribute_name>='<string value>']` exact match
- `[<attribute_name>^='<string value>']` starts with
- `[<attribute_name>$='<string value>']` ends with
- `[<attribute_name>~='<string value>']` contains word
- `[<attribute_name>*='<string value>']` contains substring

```suggestion
    app$get_text("#shiny-modal > div > div > div.modal-body > div > button[data-dismiss]")
```

or (with a selector that is less specific but resilient to any changes in modal internal structure)

```suggestion
    app$get_text("#shiny-modal div.modal-footer button[data-dismiss]")
```

I prefer the last one, but it's up to you

Reply:
Wow, you are the true legend. Thanks for the explanations. Will save them locally in my notes